src/java.desktop/share/classes/com/sun/imageio/plugins/common/ReaderUtil.java

```diff
@@ -258,4 +258,28 @@ public static byte[] staggeredReadByteStream(ImageInputStream iis,
         }
         return decodedData;
     }
+
+    /**
+     * Tries to read {@code b.length} bytes from the stream,
+     * and stores them into {@code b} starting at index 0.
+     * If the end of the stream is reached, a {@code false}
+     * will be returned.
+     *
+     * @param  iis  the stream to read.
+     * @param  b    an array where to store the {@code byte}s.
+     * @return {@code true} on success, or {@code false} on EOF.
+     */
+    public static boolean tryReadFully(ImageInputStream iis, byte[] b)
+        throws IOException
+    {
+        int offset = 0;
+        do {
+            int n = iis.read(b, offset, b.length - offset);
+            if (n < 0) {
+                return false;       // EOF
+            }
+            offset += n;
+        } while (offset < b.length);
+        return true;
+    }
```

Comment:
Any reason not to catch EOFException exception in the exceptional situation?

Reply:
No strong reason. I thought it was safer because a -1 return value means that the stream reached EOF on normal condition, while `EOFException` could still be a problem with the stream for example if the stream wraps another `InputStream` for applying decompression on-the-fly (e.g. a ZIP file), and the compressed stream is truncated before the end of the compression chunk.